CMakeLists.txt

```diff
@@ -132,6 +132,11 @@ if(ENABLE_BOOST)
   endif()
 endif()
 
+# if this flag is set, Eigen will use ThreadpoolDevice with the number of thread as specified
+if (THREADPOOL) 
```

Comment:
So overall it will use the thread-pool functionality of Eigen code, Right? Could you please add some tests and show sort of the performance gain after opening this? If that is the case, could we open it by default when using Eigen + GPU, or so? If I am a user, I may feel uncertain about the relationship between `-DBACKEND=cuda` with `-DTHREAD_POOL`.